authority/tls.go

```diff
@@ -77,19 +77,22 @@ func (a *Authority) Sign(csr *x509.CertificateRequest, signOpts provisioner.Opti
 		}
 	}
 
-	stepCSR, err := stepx509.ParseCertificateRequest(csr.Raw)
-	if err != nil {
-		return nil, nil, &apiError{errors.Wrap(err, "sign: error converting x509 csr to stepx509 csr"),
-			http.StatusInternalServerError, errContext}
+	if err := csr.CheckSignature(); err != nil {
```

Comment:
aren't we already doing this here? https://github.com/smallstep/certificates/blob/master/api/api.go#L180-L182

Reply:
Yes, but I thought it makes sense to do it internally. For example, if we add a gRPC frontend and we forget to do it in the gRPC api.